src/main/java/mezz/jei/plugins/jei/info/IngredientInfoRecipe.java

```diff
@@ -90,4 +89,68 @@ private IngredientInfoRecipe(List<T> ingredients, IIngredientType<T> ingredientT
 	public List<T> getIngredients() {
 		return ingredients;
 	}
+
+	private static class ExpandNewLineTextAcceptor implements ITextProperties.IStyledTextAcceptor<Void> {
+
+		private final List<ITextProperties> lines = new ArrayList<>();
+		@Nullable
+		private IFormattableTextComponent lastComponent;
+
+		@Override
+		public Optional<Void> accept(Style style, String line) {
```

Comment:
I think this should have unit tests if possible, the rules for it are pretty complicated.

Reply:
Unfortunately the cases where the rules are complex (when lastComponent is being modified) is only really used in translation text components when doing parameter replacement and has code that ensures the game is running and the language map is initialized which makes this rather complicated to do unit tests for until 1.17 when it will hopefully become easier as from what I understand Mojang is exposing their testing suite.